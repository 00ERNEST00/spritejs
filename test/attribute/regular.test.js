import RegularAttr from '../../src/attribute/regular';

describe('attribute set get', () => {
  const subject = {
    onPropertyChange(key, value, oldValue, attributes) { },
  };
  const attr = new RegularAttr(subject);

  test('get set', () => {
    expect(attr.d).toBe('');
    attr.radius = 10;
    expect(attr.d).toMatchSnapshot();
    attr.edges = 7;
    expect(attr.d).toMatchSnapshot();
    attr.offsetAngle = 60;
    expect(attr.d).toMatchSnapshot();
    expect(attr.closeType).toBe('normal');
  });
});