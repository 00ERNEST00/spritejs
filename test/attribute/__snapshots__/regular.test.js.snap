// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute set get get set 1`] = `"M6.123233995736766e-16 -10L8.660254037844387 4.999999999999998L-8.660254037844386 5.0000000000000036Z"`;

exports[`attribute set get get set 2`] = `"M6.123233995736766e-16 -10L7.818314824680298 -6.234898018587335L9.749279121818237 2.225209339563144L4.3388373911755815 9.009688679024192L-4.338837391175581 9.009688679024192L-9.749279121818237 2.225209339563145L-7.818314824680299 -6.234898018587334Z"`;

exports[`attribute set get get set 3`] = `"M8.660254037844386 -5L9.308737486442043 3.6534102436639495L2.947551744109043 9.555728057861407L-5.633200580636217 8.262387743159952L-9.972037971811801 0.7473009358642467L-6.801727377709197 -7.33051871829826L1.4904226617617398 -9.888308262251286Z"`;
